errors/TypesDoNotUnify.md

```diff
@@ -33,6 +33,9 @@ In the example above, the types `String` and `Int` can never be made equal, henc
 - Look carefully at the error, especially the information at the end. Usually, it will help to narrow down the offending expression. For example, in the error message above, we are told that the error occurred "while checking that expression `"test"` has type `Int`".
 
 ## Notes
+### Arrays
+
+Unlike in Javascript, all elements of a PureScript array must be the same type. For example, ```[true, false]``` or ```[1, 2, 3]``` work fine but ```[1, false]``` does not.
```

Comment:
How would you explain type unification here? Something like: "This fails because the compiler attempts to unify this type as follows: The outer-most type is `Array a`, so the type-checker finds `a` by using the type of its contents. It finds the contents' type by checking the type of each elements, and reducing the results to a single value. It happens to read the type of the last element first, which is `Boolean` for `false`, using that as `a`, then verifying all elements are `Boolean`. This type-check fails when reading `Int` for `1`, so throws the error:

```
  Could not match type
           
    Boolean
           
  with type
       
    Int
```